pkg/auth/serviceaccount.go

```diff
@@ -94,3 +97,30 @@ func EnsureOIDCServiceAccountExistsForResource(ctx context.Context, serviceAccou
 
 	return nil
 }
+
+type OIDCStatusMarker interface {
+	MarkOIDCIdentityCreatedSucceeded()
+	MarkOIDCIdentityCreatedSucceededWithReason(reason, messageFormat string, messageA ...interface{})
+	MarkOIDCIdentityCreatedFailed(reason, messageFormat string, messageA ...interface{})
+	MarkStatus(authStatus *duckv1.AuthStatus)
+}
+
+func OIDCAuthStatusUtility(ctx context.Context, serviceAccountLister corev1listers.ServiceAccountLister, kubeclient kubernetes.Interface, gvk schema.GroupVersionKind, objectMeta metav1.ObjectMeta, marker OIDCStatusMarker) pkgreconciler.Event {
```

Comment:
Can we find a better name, as `OIDCAuthStatusUtility` is not very clear IMO. Maybe something like `SetupOIDCServiceAccount` :shrug: